### PR TITLE
scheduler: migrate test framework to testify

### LINKED ISSUE #5121: migrate test-infra to testify for `server/schedulers`

### PR DESCRIPTION
**Pull request by HunDunDM**
### What problem does this PR solve?

Issue Number: Close #5121

### Check List

Tests

- Unit test

### Release note

```release-note
None
```

### CODE REVIEW

**Review by AndreMouche — APPROVED** [review]
LGTM

**Review by bufferflies — APPROVED** [review]

File: pkg/testutil/operator_check.go [hunk 1]
```diff
@@ -64,61 +63,57 @@ func trimTransferLeaders(op *operator.Operator) (steps []operator.OpStep, lastLe
...
-	c.Assert(steps[0].(operator.AddLearner).ToStore, check.Equals, targetID)
-	c.Assert(steps[1], check.FitsTypeOf, operator.PromoteLearner{})
-	c.Assert(steps[2].(operator.RemovePeer).FromStore, check.Equals, sourceID)
-	c.Assert(lastLeader, check.Not(check.Equals), uint64(0))
-	c.Assert(lastLeader, check.Not(check.Equals), sourceID)
+	re.Len(steps, 3)
+	re.Equal(targetID, steps[0].(operator.AddLearner).ToStore)
+	re.IsType(operator.PromoteLearner{}, steps[1])
+	re.Equal(sourceID, steps[2].(operator.RemovePeer).FromStore)
+	re.NotEqual(uint64(0), lastLeader)
```
Comment (lhy1024) [inline pkg/testutil/operator_check.go:98]: how about `notZero`

File: server/schedulers/scheduler_test.go [hunk 2]
```diff
@@ -538,12 +505,12 @@ func (s *testBalanceLeaderSchedulerWithRuleEnabledSuite) TestBalanceLeaderWithCo
 	}
 
 	for _, testCase := range testCases {
-		c.Logf(testCase.name)
```
Comment (lhy1024) [inline server/schedulers/scheduler_test.go:541]: why remove it?

**Review by lhy1024 — COMMENTED** [review]
the rest LGTM

**Review by lhy1024 — APPROVED** [review]

**Pull request merged**